### ISSUE TITLE
No function signatures available for --only map_keys

### ISSUE THREAD

**Issue by mbasmanova** [facebookincubator/velox#3533]
### Bug description

Expression fuzzer currently fails if --only flag lists only generic functions, e.g. --only map_keys

This prevents us from testing many functions in isolation.

CC: @kgpai @bikramSingh91 @kagamiori 

```
velox_expression_fuzzer_test --only map_keys --velox_fuzzer_enable_complex_types --logtostderr=1 --duration_sec 60

I20221216 06:58:55.322088 2765706 FuzzerRunner.h:120] Skipping special form: if
I20221216 06:58:55.323611 2765706 FuzzerRunner.h:120] Skipping special form: coalesce
I20221216 06:58:55.323622 2765706 FuzzerRunner.h:120] Skipping special form: switch
I20221216 06:58:55.324132 2765706 ExpressionFuzzer.cpp:351] Total candidate functions: 1 (1 signatures)
I20221216 06:58:55.324153 2765706 ExpressionFuzzer.cpp:355] Functions with at least one supported signature: 1 (100.00%)
I20221216 06:58:55.324172 2765706 ExpressionFuzzer.cpp:359] Functions with no supported signature: 0 (0.00%)
I20221216 06:58:55.324182 2765706 ExpressionFuzzer.cpp:363] Supported function signatures: 1 (100.00%)
I20221216 06:58:55.324190 2765706 ExpressionFuzzer.cpp:367] Unsupported function signatures: 0 (0.00%)
E20221216 06:58:55.324229 2765706 Exceptions.h:68] Line: /Users/mbasmanova/cpp/velox-1/velox/expression/tests/ExpressionFuzzer.cpp:778, Function:go, Expression: !signatures_.empty() No function signatures available., Source: RUNTIME, ErrorCode: INVALID_STATE
libc++abi: terminating with uncaught exception of type facebook::velox::VeloxRuntimeError: Exception: VeloxRuntimeError
Error Source: RUNTIME
Error Code: INVALID_STATE
Reason: No function signatures available.
Retriable: False
Expression: !signatures_.empty()
Function: go
File: /Users/mbasmanova/cpp/velox-1/velox/expression/tests/ExpressionFuzzer.cpp
Line: 778
```

### System information

n/a

### Relevant logs

```bash
n/a
```

**Comment by darrenfu** [comment]
Also got affected for a few map functions, such as map_from_entries.

**Comment by bikramSingh91** [comment]
this will be fixed as a part of #3388. It happens because the current code relies on function signatures to pick the starting return type and currently it only picks from the non-templatized function signatures. Therefore it requires "signatures_" to be non empty and has a VELOX_CHECK for that.

The part in #3388 that deals with using random top level return type (using VectorFuzzer::randType() ) will fix this.

**Comment by mbasmanova** [comment]
@bikramSingh91 Bikram, what is the timeline for #3388? If it is more than a week away, can we fix this issue first to unblock folks adding new functions?

**Comment by kagamiori** [comment]
I think we can do a quick fix to unblock the new functions before Bikram implement a general solution. Basically we can flip the coin to determine whether to choose a function from signatures_ or signatureTemplates_. If the chosen one is empty, then we choose from the other.

**Comment by bikramSingh91** [comment]
> I think we can do a quick fix to unblock the new functions before Bikram implement a general solution. Basically we can flip the coin to determine whether to choose a function from signatures_ or signatureTemplates_. If the chosen one is empty, then we choose from the other.

> @bikramSingh91 Bikram, what is the timeline for #3388? If it is more than a week away, can we fix this issue first to unblock folks adding new functions?

Yes, it's a minor change as we can switch to using VectorFuzzer::randType() to generate a random top level return type. Will publish a PR in a few mins.

**Comment by bikramSingh91** [comment]
> I think we can do a quick fix to unblock the new functions before Bikram implement a general solution. Basically we can flip the coin to determine whether to choose a function from signatures_ or signatureTemplates_. If the chosen one is empty, then we choose from the other.

Sounds good! let me know if you have already started on this, otherwise feel free to assign it to me.

**Comment by kagamiori** [comment]
> > I think we can do a quick fix to unblock the new functions before Bikram implement a general solution. Basically we can flip the coin to determine whether to choose a function from signatures_ or signatureTemplates_. If the chosen one is empty, then we choose from the other.
> 
> Sounds good! let me know if you have already started on this, otherwise feel free to assign it to me.

Hi Bikram, Yeah, I'm working on this. It would be great if you could help review it.

**Comment by mbasmanova** [comment]
@kagamiori Wei, Bikram, thank you for working on this. I wonder if it makes sense to start by randomly picking up a signature or signature template instead of picking up return type.

**Comment by kagamiori** [comment]
> @kagamiori Wei, Bikram, thank you for working on this. I wonder if it makes sense to start by randomly picking up a signature or signature template instead of picking up return type.

Yes, I think in #3540 we pick up a signature or signature template and use its return type as the top-level result type.

**Comment by kagamiori** [comment]
#3540 is blocked by fuzzer test failures such as #3542. It may be a problem in the fuzzer itself (e.g., simplified path). I'm going to investigate.

**Comment by mbasmanova** [comment]
@kagamiori Wei, is this done now?

**Comment by kagamiori** [comment]
> @kagamiori Wei, is this done now?

I believe so. Closing this issue since it's fixed in #3540
